test/stdlib/Random.swift

```diff
@@ -5,6 +5,31 @@ import StdlibUnittest
 
 let RandomTests = TestSuite("Random")
 
+// _stdlib_random
+
+RandomTests.test("_stdlib_random") {
+  let buffer1 = UnsafeMutableRawBufferPointer.allocate(byteCount: 8, alignment: 8)
+  let buffer2 = UnsafeMutableRawBufferPointer.allocate(byteCount: 8, alignment: 8)
+  
+  defer {
+    buffer1.deallocate()
+    buffer2.deallocate()
+  }
+  
+  // buffer1 == buffer2
+  expectTrue(buffer1.elementsEqual(buffer2))
```

Comment:
```swift
for count in [100, 1000] {
    // Initialize two large buffers.
    var bytes1 = [UInt8](repeating: 0, count: count)
    var bytes2 = [UInt8](repeating: 0, count: count)
    expectEqual(bytes1, bytes2)

    // Fill with random bytes.
    bytes1.withUnsafeMutableBytes({ _stdlib_random($0) })
    bytes2.withUnsafeMutableBytes({ _stdlib_random($0) })
    expectNotEqual(bytes1, bytes2)

    // Test for unfilled repeating zeros.
    var zeros = [UInt8](repeating: 0, count: 10)
    expectNil(memmem(&bytes1, bytes1.count, &zeros, zeros.count))
    expectNil(memmem(&bytes2, bytes2.count, &zeros, zeros.count))
}
```

I've used `memmem` in this example, but is there a stdlib API to find a subsequence?

Reply:
Here's a unit test which also shows how `_stdlib_random` might work in the Foundation overlay.

```swift
import Foundation

extension Data {
  static func _random(count: Int) -> Data {
    var result = Data(count: count)
    result.withUnsafeMutableBytes {
      (start: UnsafeMutablePointer<UInt8>) -> Void in
      let start = UnsafeMutableRawPointer(start)
      let bytes = UnsafeMutableRawBufferPointer(start: start, count: count)
      _stdlib_random(bytes)
    }
    return result
  }
}

let zeros = Data(count: 10)
for count in [100, 1000] {
  let data1 = Data._random(count: count)
  let data2 = Data._random(count: count)
  expectNil(data1.range(of: zeros))
  expectNil(data2.range(of: zeros))
  expectNotEqual(data1, data2)
}
```

Should it actually be `_swift_stdlib_random` with the `SWIFT_RUNTIME_STDLIB_SPI` attribute?

Reply:
We can keep it `_stdlib_random` with `SWIFT_RUNTIME_STDLIB_INTERNAL` atm because I plan to follow up with another proposal to fill buffers. The solution atm isn’t optimal (calling `Random.default.next()` many times), but it lets us keep stdlib visibility. In your memmem example, couldn’t you call `elementsEqual(_:)` because you’re checking if all elements are equal in sequential order?

Reply:
* [`Sequence.elementsEqual`](https://github.com/apple/swift/blob/8ef98b8dd2dd4519c2a94f491c68f1254daa4ac3/stdlib/public/core/SequenceAlgorithms.swift#L273-L345) will iterate both sequences, even if they have a different number of elements.

* [`Array.==`](https://github.com/apple/swift/blob/8ef98b8dd2dd4519c2a94f491c68f1254daa4ac3/stdlib/public/core/Arrays.swift.gyb#L2198-L2263) returns early if the `count` is different, or if the buffer identity is the same.

* [`Data.==`](https://github.com/apple/swift/blob/f38192d7b7afe08a2e26cf8077769183cd18d063/stdlib/public/SDK/Foundation/Data.swift#L1844-L1870) also returns early, and uses the `memcmp` API (which might be faster).

In this example, the arrays have the same number of elements in different buffers, so `elementsEqual` and `==` will do the same thing.

The `memmem` API might not be available on all platforms.